flink-runtime/src/main/java/org/apache/flink/runtime/state/changelog/StateChangelogStorageFactory.java

```diff
@@ -45,4 +51,16 @@ StateChangelogStorage<?> createStorage(
 
     /** Create the storage for recovery. */
     StateChangelogStorageView<?> createStorageView(Configuration configuration) throws IOException;
+
+    /** Extract the relevant to this factory portion of the configuration. */
+    default Configuration extractConfiguration(ReadableConfig src) {
```

Comment:
IIUC, Even if users haven't setted these parameter in job level, we will still overide them in current logic?
Should we just overide them when users have setted them ?